app/models/course/lesson_plan/item.rb

```diff
@@ -65,14 +65,16 @@ class Course::LessonPlan::Item < ApplicationRecord
   scope :with_personal_times_for, (lambda do |course_user|
     personal_times =
       if course_user.nil?
-        []
+        nil
       else
-        Course::PersonalTime.where(course_user_id: course_user.id, lesson_plan_item_id: all).to_a
+        Course::PersonalTime.where(course_user_id: course_user.id, lesson_plan_item_id: all)
```

Comment:
Why change the function signature here?

Reply:
it's because the previous one that we are using (`ManualPreloader`) is no longer compatible with the `Preloader` object that's used inside `Active Record` in Rails 7, and after checking the object itself, the execution call of `ManualPreloader` only calls `super`, means that it's actually doing the same thing as `ActiveRecord::Association::Preloader`.

Hence, I decided to just use `ActiveRecord::Association::Preloader` here instead, and with it comes the change in the argument signature that I need to use. One of them being needing to provide the `scope` variable inside the `Preloader`, in which after I reviewed about how the `scope` should be defined in our codebase, usually does not have `to_a` and that's why I changed it

(also, the default value of `scope` should it not be provided is merely `nil`, not `[]`)

Reply:
https://github.com/Coursemology/coursemology2/pull/7427/commits/e88c34b157737fd011b290b509fa1ae1475a6ca5
^ commit should be squashed with current commit.

Reply:
Done